feat: track posthog events

r4zendev opened the pull request:
## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced analytics tracking across the application with dedicated page tracking and event logging for user authentication and account creation.
  - Improved the login process by adding a password field and additional validations for better security.

- **Refactor**
  - Streamlined provider components to offer a more cohesive initialization of analytics features.

- **Chores**
  - Updated dependencies to support the new analytics and authentication enhancements.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced analytics tracking across the application with dedicated page tracking and event logging for user authentication and account creation.
  - Improved the login process by adding a password field and additional validations for better security.
  - Added a new `PostHogProvider` component for improved analytics integration.
  - Introduced the `PostHogPageView` component for tracking page views.
  - Implemented the `AnalyticsModule` for centralized analytics functionality.
  - Enhanced the `DocumentTracker` component's layout for better visual presentation.
  - Added a new method in `UnifiedApiClient` for entity matching via AI.

- **Bug Fixes**
  - Removed the document upload option from the `DocumentTrackerItemOptions` component.

- **Refactor**
  - Streamlined provider components to offer a more cohesive initialization of analytics features.
  - Updated the `AuthController`, `CustomerService`, and `UserService` to integrate analytics tracking.
  - Simplified data extraction in the `useDocumentBlocks` hook.
  - Transitioned to asynchronous handling in the rule engine for improved performance.

- **Chores**
  - Updated dependencies to support the new analytics and authentication enhancements.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

r4zendev committed:
track page views

tomer-shvadron committed:
Merge branch 'dev' into bal-3605

tomer-shvadron committed:
Merge branch 'dev' into bal-3605

tomer-shvadron committed:
chore(monitoring): remove unused posthog import

- Eliminate unused import of posthog-js in initialize-monitoring.ts
- Update subproject commit hash in data-migrations

tomer-shvadron committed:
Merge branch 'dev' into bal-3605

tomer-shvadron committed:
Merge branch 'dev' into bal-3605

tomer-shvadron committed:
feat(auth): integrate analytics service for user login events

- Add AnalyticsModule to track user login and magic link login events
- Update AuthController to use analytics service for event tracking
- Include necessary dependencies in package.json for analytics functionality

tomer-shvadron committed:
refactor(auth): update analytics event names to use EventNamesMap

- Replace EventNames with EventNamesMap for consistency
- Adjust method tracking for user login events
- Update relevant imports and constructor parameter names across services

tomer-shvadron committed:
feat: pnpm-lock update

tomer-shvadron committed:
Merge branch 'dev' into bal-3605

tomer-shvadron committed:
feat(tests): add AnalyticsModule to common testing modules

- Include AnalyticsModule for enhanced testing capabilities
- Ensure global availability of necessary services in tests

tomer-shvadron committed:
feat(tests): add analytics service to integration tests

- Integrate AnalyticsService into collection-flow integration tests
- Update dependencies in collection-flow controller tests to include AnalyticsService
- Remove unused variable workflowDefinitionRepository in workflow definition service tests

tomer-shvadron committed:
Merge branch 'dev' into bal-3605

alonp99 commented on services/workflows-service/src/customer/customer.service.ts:
```diff
@@ -72,6 +72,11 @@ export class CustomerService {
         await retry(() => new UnifiedApiClient().createCustomer(customer));
       }
 
+      this.analyticsService.track({
```
let's add a flag to differentiate if it's a demo account'

tomer-shvadron committed:
feat(analytics): update customer created event with demo account flag

- Modify AnalyticsEvents type to include isDemoAccount property
- Update CustomerService to pass demo account status in analytics event
- Update subproject commit reference in data-migrations folder

tomer-shvadron committed:
Merge branch 'dev' into bal-3605
